05/app.js

```diff
@@ -1,53 +1,75 @@
 const stats = {
-    paragraphs: {
-        'p1': 0,
-    },
-    links: {
-        '/dolor.html': 0,
-    }
-};
+	paragraphs: {
+		p1: 0,
+	},
+	links: {
+		'/dolor.html': 0,
+	},
+}
 
 /* tutaj umieść swój kod */
 
+const paragraphsList = document.querySelectorAll('p')
+
+const calculateClicks = function (element) {
+	element.addEventListener('click', function (e) {
+		if (e.target.tagName === 'P') {
+			if (typeof stats.paragraphs[e.target.dataset.id] === 'undefined') {
+				stats.paragraphs[e.target.dataset.id] = 0
+			}
+			stats.paragraphs[e.target.dataset.id] = stats.paragraphs[e.target.dataset.id] + 1
+		} else if (e.target.tagName === 'A') {
+			e.preventDefault()
+			if (typeof stats.links[e.target.getAttribute('href')] === 'undefined') {
+				stats.links[e.target.getAttribute('href')] = 0
+			}
+			stats.links[e.target.getAttribute('href')] = stats.links[e.target.getAttribute('href')] + 1
```

Comment:
Zwróć uwagę, że sporo dzieje się tutaj takich samach działań tj.
1) zerujemy licznik jeśli element nie istnieje o podanym indeksie
2) zwiększam o jeden wartość podanego indeksu

Może dałoby sie napisać funkcję, która wykonuje te czynności względem wskazanego indeksu? np. `changeStats(propName, index)` - traktuj to jako zadanie dodatkowe ;)